core/src/main/scala/Logic.scala

```diff
@@ -13,58 +13,57 @@ class CarApi(carService: CarService):
       input.split(" ", 3).toList match
         case List(f1, f2, f3) =>
           val car = Car(f1, f2, f3)
-          carService.register(car).map(_ => "OK: Car registered").catchAll {
+          carService.register(car).as("OK: Car registered").catchAll {
             case _: LicensePlateExistsError =>
               ZIO
                 .logError(
                   s"Cannot register: $car, because a car with the same license plate already exists"
                 )
-                .map(_ => "Bad request: duplicate")
+                .as("Bad request: duplicate")
             case t =>
               ZIO
                 .logErrorCause(s"Cannot register: $car, unknown error", Cause.fail(t))
-                .map(_ => "Internal server error")
+                .as("Internal server error")
           }
-        case _ => ZIO.logError(s"Bad request: $input").map(_ => "Bad Request")
+        case _ => ZIO.logError(s"Bad request: $input").as("Bad Request")
     }
 
 object CarApi:
-  val live: ZLayer[CarService, Any, CarApi] = ZLayer.fromFunction(CarApi(_))
+  val live: ZLayer[CarService, Nothing, CarApi] = ZLayer.fromFunction(CarApi(_))
+  def register(input: String): ZIO[CarApi, Nothing, String] =
```

Comment:
as  mentioned in the comments for #1, I'd like to keep the usage of environments for the purposes specified in the blog; here that would be another category in which we're using the env, and it's not quite clear for me what would be the "best practices" here